hoomd/mpcd/CollisionMethod.cc

```diff
@@ -102,11 +119,17 @@ void mpcd::CollisionMethod::collide(uint64_t timestep)
         if (m_exec_conf->isCUDAEnabled())
             {
             storeInitialEmbeddedGroupVelocitiesGPU(timestep);
+            // thermalize constituent particles to add energy to the rigid body
+            drawVelocitiesConstituentParticlesGPU(timestep);
+            getNetVelocityRigidBodyGPU(timestep);
+            applyThermalizedVelocityVectorsGPU(timestep);
             }
         else
 #endif
             {
             storeInitialEmbeddedGroupVelocities(timestep);
+            // thermalize constituent particles to add energy to the rigid body
+            thermalizeConstituentParticles(timestep);
```

Comment:
Is this happening in the wrong order? I think we need to store the embedded velocities *after* thermalization so we know how much momentum is transferred by the collision, not by thermalization + collision. Am I thinking about that right?

Reply:
Since the thermalization is meant to be adding kinetic energy and momentum to the constituent particles (but not net momentum when taking it all together) then the net change of a single constituent particle would be final velocity - velocity that matches the rigid body's center of mass velocity and angular momentum prior to the collision step. It doesn't make much sense to me to use the thermalized velocities as the initial velocities instead. 

I may test switching the order first because I'm not sure that would come to the right temperature.

Reply:
I tested it, and using the 810 mass sphere it does come out to the same temperature.